packages/office-ui-fabric-react/src/utilities/positioning/positioning.ts

```diff
@@ -263,6 +263,32 @@ function _flipToFit(
   };
 }
 
+/**
+ * Flips only the alignment edge of an element rectangle. This is used instead of nudging the alignment edges into position,
+ * when alignPerfectlyWithTarget is specified.
+ * Returns the initial elementEstimate if the flipping does not work.
+ * @param elementEstimate
+ * @param target
+ * @param bounding
+ * @param gap
+ */
+function _flipAlignmentEdgeOnly(elementEstimate: IElementPosition, target: Rectangle, bounding: Rectangle, gap: number): IElementPosition {
+  const { alignmentEdge, targetEdge, elementRectangle } = elementEstimate;
+  if (alignmentEdge !== undefined) {
+    const newEstimate = _estimatePosition(elementRectangle, target, { targetEdge: targetEdge, alignmentEdge: alignmentEdge * -1 }, gap);
+
+    if (_isEdgeInBounds(newEstimate, bounding, alignmentEdge) && _isEdgeInBounds(newEstimate, bounding, alignmentEdge * -1)) {
```

Comment:
I don't think we should even check if it's in bounds, we're moving it because it's already not in bounds either way it won't fit so it doesn't hurt to keep this method simple and just return the flip. Or you could check this in the method that calls it and decide if you want to set it as the return value there.

Reply:
Changed the function to now just swap the alignments.